cfme/tests/test_db_restore.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+
+import fauxfactory
+import pytest
+
+from cfme.configure import configuration as config
+from cfme.cloud import provider as cloud_provider
+from cfme.infrastructure.provider import wait_for_a_provider
+from cfme.infrastructure.virtual_machines import Vm
+from utils.appliance import provision_appliance
+from utils.log import logger
+from utils.providers import setup_a_provider
+from utils import conf, testgen, version
+
+pytest_generate_tests = testgen.generate(testgen.infra_providers, scope="module")
+
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = testgen.provider_by_type(
+        metafunc, ['virtualcenter'], choose_random=True)
+    if not idlist:
+        return
+    metafunc.parametrize(argnames, argvalues, ids=idlist, scope='module')
+
+try:
+    server_roles_conf = conf.cfme_data.server_roles
+except KeyError:
+    server_roles_conf = {
+        'all': [],
+        'sets': {},
+    }
```

Comment:
Can this logic be moved inside the all_possible_roles fixture since that is the only place where it is used? I don't quite like pulling config data on module import.